br/pkg/utils/key.go

```diff
@@ -89,3 +95,29 @@ func CompareEndKey(a, b []byte) int {
 
 	return bytes.Compare(a, b)
 }
+
+// encode
+func FormatAPIV2Key(key []byte, isEnd bool) []byte {
+	if isEnd && len(key) == 0 {
+		return []byte{APIV2KeyPrefixEnd}
+	}
+	apiv2Key := []byte{APIV2KeyPrefix}
+	return append(apiv2Key, key...)
+}
+
+// conversion between formated APIVersion key and backupmeta key
+// for example, backup apiv1 -> apiv2, add `r` prefix and `s` for empty end key.
+// apiv2 -> apiv1, remove first byte. This is useful when decode the response keys.
+func ConvertBackupConfigKey(backupKey []byte, isEnd bool, srcAPIVer kvrpcpb.APIVersion, dstAPIVer kvrpcpb.APIVersion) []byte {
```

Comment:
How about define a struct as `Range`, and do the codec, and cross api version conversion as a whole ? Then the caller do not need to care about the `isEnd`.

Besides, additional to backup meta, this conversion function can be used in other place. E.g., `cfg.adjustBackupRange` can be treated as a conversion from v1 to source api version.

Reply:
good idea.

Reply:
done.